### PR TITLE
Update mongo sample to koop 3.x with mongo updates

### PR DESCRIPTION
**Pull request by phpmaps**
@dmfenton @jgravois - let me know if this is any good for merging.  It's uses koop 3.5.0 and makes use of Mongo's [readable stream]([http://mongodb.github.io/node-mongodb-native/2.0/api/Cursor.html#stream).  Also connection details are in the config.

### CODE REVIEW

File: mongo.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+const config = require('config')
+const MongoClient = require('mongodb')
+
+module.exports = function (koop) {
+
+  this.getData = (req, callback) => {
+    var url = config.get('mongo.connectionUrl')
+    MongoClient.connect(url, (err, db) => {
+      var features = []
+      var collection = db.collection('restaurants')
```
Comment (dmfenton) [inline mongo.js]: this is a good place to generalize by using `req.params.id`

File: mongo.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      var stream = collection.find({}).stream()
+      stream.on('data', (doc) => {
+        features.push(this.formatFeature(doc))
```
Comment (dmfenton) [inline mongo.js]: makes sense to translate the features as they come in, rather than waiting for the full batch

File: mongo.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      })
+      stream.on('end', ()=>{
+        var geojson = this.translate(features)
+        callback(null, geojson) // hand the geojson back to Koop
+        db.close()
```
Comment (dmfenton) [inline mongo.js]: I would close the DB connection immediately after the stream ends

**Comment by phpmaps** [comment]
@dmfenton - ok changes made per code review.  much more abstract now by implementing ``` req.params.id ``` .

File: index.js [hunk 4]
```diff
@@ -1,8 +1,14 @@
+// the name of provider is used by koop to help build default routes for FeatureService and a preview
+const pkg = require('./package.json')
+const provider = {
+  name: 'mongo',
+  hosts: false,
+  disableIdParam: true,
```
Comment (dmfenton) [inline index.js:6]: this needs to be set to `false` or deleted

**Review by dmfenton — CHANGES_REQUESTED** [review]
just need to change one thing from what I can tell

**Comment by rgwozdz** [comment]
closing

**Pull request closed without merging**